File: src/main/java/com/pulsesocial/service/MessageService.java
package com.pulsesocial.service;

import com.pulsesocial.dto.MessageResponse;
import com.pulsesocial.dto.SendMessageRequest;
import com.pulsesocial.dto.UserResponse;
import java.util.List;

public interface MessageService {
    MessageResponse sendMessage(SendMessageRequest request);
    List<MessageResponse> getConversation(Long otherUserId);
    List<UserResponse> getRecentConversations();
    MessageResponse updateMessage(Long id, String newContent);
    String deleteMessage(Long id);
}

File: PulseSocial-Frontend/src/services/messageService.js
import api from "../api/axios";

export const sendMessage = (data) => {
    return api.post("/messages", data);
};

export const getConversation = (otherUserId) => {
    return api.get(`/messages/conversation/${otherUserId}`);
};

export const getRecentConversations = () => {
    return api.get("/messages/conversations");
};

export const updateMessage = (id, content) => {
    return api.put(`/messages/${id}`, { content });
};

export const deleteMessage = (id) => {
    return api.delete(`/messages/${id}`);
};
